Composer/packages/client/src/pages/design/createDialog.tsx

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import React, { useState, useContext } from 'react';
+import formatMessage from 'format-message';
+import { PrimaryButton, DefaultButton } from 'office-ui-fabric-react/lib/Button';
+import { DialogFooter } from 'office-ui-fabric-react/lib/Dialog';
+import { Stack, StackItem } from 'office-ui-fabric-react/lib/Stack';
+import { TextField } from 'office-ui-fabric-react/lib/TextField';
+
+import { DialogCreationCopy } from '../../constants';
+import { DialogWrapper } from '../../components/DialogWrapper/index';
+import { StorageFolder } from '../../store/types';
+import { StoreContext } from '../../store';
+
+import { name, description, styles as wizardStyles } from './styles';
+
+interface CreateDialogParams {
+  name: string;
+  description: string;
+}
+
+interface CreateDialogProps {
+  onSubmit: (createDialogParams: CreateDialogParams) => void;
+  onDismiss: () => void;
+  onCurrentPathUpdate?: (newPath?: string, storageId?: string) => void;
+  focusedStorageFolder?: StorageFolder;
+  currentPath?: string;
+  isOpen: boolean;
+}
+
+const initialFormDataError = {};
```

Comment:
why we define this initial state outside the component? it could be written inline )
```ts
useState({})
```